vital/util/ring_buffer.h

```diff
@@ -0,0 +1,138 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief Base interface for simple buffer classes
+ */
+
+#ifndef KWIVER_VITAL_UTIL_RING_BUFFER_H_
+#define KWIVER_VITAL_UTIL_RING_BUFFER_H_
+
+#include <vector>
+#include <exception>
+
+#include "buffer.h"
+
+namespace kwiver {
+namespace vital {
+
+
+template< class Data >
+class ring_buffer : public buffer< Data >
```

Comment:
I see very little reason for having this class.  I suggest we just use an `std::deque` instead of a ring buffer and keep the max_capacity in a separate side car variable.